### ISSUE TITLE
Can't install with docker on M1 Mac

### ISSUE THREAD

**Issue by genesiscz** [confetti-framework/confetti#100]
```
Pulling mysql (mysql:8.0)...
8.0: Pulling from library/mysql
ERROR: no matching manifest for linux/arm64/v8 in the manifest list entries
```

Simply replacing 'mysql:8.0' to 'mysql/mysql-server:8.0' seem to work. But I get

```
Status: Downloaded newer image for mysql/mysql-server:8.0
Pulling redis (redis:alpine)...
alpine: Pulling from library/redis
2914792bc417: Pull complete
5ce28e8636b2: Pull complete
393e448a41a0: Pull complete
b2da15e8aaed: Pull complete
72d76ccc5dc4: Pull complete
ed505bb98182: Pull complete
Digest: sha256:d2b1b0bf521f55381c81b1bc0218aec4ddc92fb545e9e8492e8ff324462eed58
Status: Downloaded newer image for redis:alpine
Building confetti.local

Traceback (most recent call last):
  File "compose/cli/main.py", line 67, in main
  File "compose/cli/main.py", line 126, in perform_command
  File "compose/cli/main.py", line 1070, in up
  File "compose/cli/main.py", line 1066, in up
  File "compose/project.py", line 615, in up
  File "compose/service.py", line 362, in ensure_image_exists
  File "compose/service.py", line 1147, in build
compose.service.BuildError: (<Service: confetti.local>, {'message': 'Cannot locate specified Dockerfile: Dockerfile'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "docker-compose", line 3, in <module>
  File "compose/cli/main.py", line 78, in main
TypeError: can only concatenate str (not "dict") to str
[76334] Failed to execute script docker-compose
```
Dockerfile is in place. What now?

**Comment by NWBY** [comment]
Looks like this could be a known issue on M1.
Are you able to try changing this to use MariaDB?
```
image: mariadb:10.5.8
```

**Comment by genesiscz** [comment]
```
Building confetti.local

Traceback (most recent call last):
  File "compose/cli/main.py", line 67, in main
  File "compose/cli/main.py", line 126, in perform_command
  File "compose/cli/main.py", line 1070, in up
  File "compose/cli/main.py", line 1066, in up
  File "compose/project.py", line 615, in up
  File "compose/service.py", line 362, in ensure_image_exists
  File "compose/service.py", line 1147, in build
compose.service.BuildError: (<Service: confetti.local>, {'message': 'Cannot locate specified Dockerfile: Dockerfile'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "docker-compose", line 3, in <module>
  File "compose/cli/main.py", line 78, in main
TypeError: can only concatenate str (not "dict") to str
[42854] Failed to execute script docker-compose

```

**Comment by bosix** [comment]
Hi @genesiscz 

it seems that docker-compose cannot find the Dockerfile (see message: `Cannot locate specified Dockerfile: Dockerfile`). Please check that you have `docker-compose.yml` and `Dockerfile` both directly inside the root directory where you execute `docker-compose up`

**Comment by nick-d-ford** [comment]
Hi Guys,
M1 Macs will try to pull arm based images by default and fail if there isn't one available. you can however tell it to use other arc images by prefixing the arch on the image name.
On an M1 Mac here and only had to change the following to get confetti up and running under docker:

docker-compose.yml:34        
`        image: 'amd64/mysql:8.0'`

Dockerfile:1
`FROM amd64/ubuntu:20.04`

After the next docker-compose up, confetti came up and homepage accessible on localhost port 80.

**Comment by reindert-vetter** [comment]
@binary-vision Thanks for your help! Can't this just be implemented? I can't think of any cons. Then I would like to make a pull-request, but you can also create a pull request yourself.

**Comment by genesiscz** [comment]
> Hi @genesiscz
> 
> it seems that docker-compose cannot find the Dockerfile (see message: `Cannot locate specified Dockerfile: Dockerfile`). Please check that you have `docker-compose.yml` and `Dockerfile` both directly inside the root directory where you execute `docker-compose up`

I said Dockerfile is in place, it is just in the same directory as `docker-compose.yml`.




> Hi Guys,
> M1 Macs will try to pull arm based images by default and fail if there isn't one available. you can however tell it to use other arc images by prefixing the arch on the image name.
> On an M1 Mac here and only had to change the following to get confetti up and running under docker:
> 
> docker-compose.yml:34
> ` image: 'amd64/mysql:8.0'`
> 
> Dockerfile:1
> `FROM amd64/ubuntu:20.04`
> 
> After the next docker-compose up, confetti came up and homepage accessible on localhost port 80.

looks like it's not a problem with mysql/mariadb as it was pulled correctly. 
This gives 

`Pulling mysql (amd64/mysql:10.5.8)...
ERROR: manifest for amd64/mysql:10.5.8 not found: manifest unknown: manifest unknown`

if I use         image: 'mariadb:10.5.8'
with Dockerifile modification, I get the same error.

```
➜  myFirstProject docker-compose up
Building confetti.local

Traceback (most recent call last):
  File "compose/cli/main.py", line 67, in main
  File "compose/cli/main.py", line 126, in perform_command
  File "compose/cli/main.py", line 1070, in up
  File "compose/cli/main.py", line 1066, in up
  File "compose/project.py", line 615, in up
  File "compose/service.py", line 362, in ensure_image_exists
  File "compose/service.py", line 1147, in build
compose.service.BuildError: (<Service: confetti.local>, {'message': 'Cannot locate specified Dockerfile: Dockerfile'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "docker-compose", line 3, in <module>
  File "compose/cli/main.py", line 78, in main
TypeError: can only concatenate str (not "dict") to str
[62837] Failed to execute script docker-compose
➜  myFirstProject ls
Dockerfile         README.md          bootstrap          docker-compose.yml go.mod             main.go            routes             test
LICENSE            app                config             go-first           go.sum             resources          storage
➜  myFirstProject 

```

**Comment by binary-vision** [comment]
Hi @genesiscz,
If you can provide steps to reproduce, I'd be happy to give it a go locally and see if I can solve your issue.

**Comment by reindert-vetter** [comment]
There must also be someone else testing this. Otherwise I have some time tonight (in 9 o'clock).

**Comment by genesiscz** [comment]
> Hi @genesiscz,
> It sounds like you have moved on to another issue here.
> If you can provide steps to reproduce, I'd be happy to give it a go locally and see if I can solve your issue.

Issue remains the same as specified in my first post. Simply replacing mysql with mysql-server worked, but didn't resolve the Dockerfile not found error.

**Comment by genesiscz** [comment]
Got it! I had `context: go-first` in my docker-compose.yml instead of `./` (but have no idea why). 

Now I got 
```
---> Running in f901810d2feb
/lib64/ld-linux-x86-64.so.2: No such file or directory
ERROR: Service 'confetti.local' failed to build : The command '/bin/sh -c go get github.com/cespare/reflex' returned a non-zero code: 255
```
while using original docker-compose.yml and Dockerfile.
So I used the tip from @binary-vision and tried `FROM amd64/ubuntu:20.04`

<img width="525" alt="image" src="https://user-images.githubusercontent.com/1512353/107368172-80241900-6ae0-11eb-9271-546287b308f9.png">

Yahoo! With maria instead of mysql, I am getting

```
mailhog_1         | [APIv1] KEEPALIVE /api/v1/events
mysql_1           | 2021-02-09 13:06:11 4 [Warning] Access denied for user 'root'@'localhost' (using password: NO)
mysql_1           | 2021-02-09 13:06:41 5 [Warning] Access denied for user 'root'@'localhost' (using password: NO)
```
which is probably the only error I've seen. So I tried to replace maria with `amd64/mysql:8.0`.

It worked without errors! So I tried to even change some stuff/create a route/controller/view and it works smoothly!

Wow! I was thinking about learning Go but this transfer from Laravel makes it so much easier. Thank you a lot for your work!

<img width="390" alt="image" src="https://user-images.githubusercontent.com/1512353/107368738-35ef6780-6ae1-11eb-9cfe-9817e97eb628.png">
